code/modules/projectiles/projectile.dm

```diff
@@ -123,23 +137,98 @@
 	return 1 //Bullets don't drift in space
 
 
-/obj/item/projectile/proc/fire()
-	spawn()
-		while(loc)
-			if(kill_count < 1)
-				qdel(src)
-				return
-			if(!paused)
+/obj/item/projectile/proc/fire(var/setAngle)
+	if(setAngle) Angle = setAngle
+	if(!legacy)
+		spawn() //New projectile system

```

Comment:
I believe ```spawn()``` is the same as ```spawn(0)```, which is to have the code in the spawn block run AS SOON AS POSSIBLE. this can be used to make things happen at the same time (think explosions, all the turfs/objects/mobs in the radius explode at the same time, because they're ```spawn(0)```'d)
Normally spawn is used for delays though.